### PR TITLE
Refactor public api

### PR DESCRIPTION
**Pull request by Frederiks96**
This PR introduces a SwiftUI-like API with modifiers instead of the closure-based configuration API.

An example of this change is the `Build` task.
#### Before
```swift
Build {
    $0.configure(
        projectType: .project("MyApp"),
        scheme: "MyApp",
        configuration: Configuration.debug,
        sdk: Sdk.iPhoneSimulator
    )
}
```
#### After
```swift
Build()
    .project("MyApp")
    .scheme("MyApp")
    .configuration(Configuration.debug)
    .sdk(Sdk.iPhoneSimulator)
```

#### Sum up
- Separated `ProjectType` into two different modifiers: `.project(_:)` and `.workspace(_:)`
  - `Archive` and `ExportArchive` are not yet converted, though
- Updated documentation files
- Refactor to SwiftUI modifier approach
- Updated dependencies

### Indentation
It seems this project has been configured with spaces - should we continue doing that? In that case, I can re-configure my editor and re-indent the files.

### Suggestions for Retry
We should consider let `Retry` take more than one task - this could be beneficial if a series of tasks work together, and if one of them fail, it needs to retry the failed one - or should it only be one task? When using `@resultBuilder`s, I don't think we can restrict the number of tasks defined in the closure, but we could maybe make an `init(times: UInt, _ builder: () -> Task)`, which is not a `@resultBuilder`, making the compiler enforce exactly **one** task returned from the closure, since that is the currently supported behaviour. The current initialiser allows for an invalid task, such as `Retry(times: 2) { }`. We should probably also log errors before retrying just for good measure.

### CODE REVIEW

**Comment by onmyway133** [comment]
@Frederiks96 Great work, thanks for the PR 👍 Looks good to me in most parts, I will add some more comments

- About indentation, I prefer 4 spaces seeing that the norm for most open source projects
- About retry, what do you think of retry as a modifier, like

```swift
Build()
    .retry(5)
```

**Comment by Frederiks96** [comment]
> @Frederiks96 Great work, thanks for the PR 👍 Looks good to me in most parts, I will add some more comments
> 
> * About indentation, I prefer 4 spaces seeing that the norm for most open source projects
> * About retry, what do you think of retry as a modifier, like
> 
> ```swift
> Build()
>     .retry(5)
> ```

@onmyway133 Thanks, I will re-indent the files shortly :+1: 

About the retry, it seems like an excellent idea! This will, however, be an exception to the rule that currently all tasks return `self` which is an instance of `Self` after each modifier. I guess `retry(5)` would then return a `Retry`, which means that this should be the last modifier.. I guess we could actually add this as an extension to `Task` with the following implementation:
```
func retry(_ times: UInt) -> Retry {
    return Retry(times: times) { self }
}
```
How do you feel about `retry(_:)` being a slightly different modifier than the rest? I guess this is okay, since there are some cases in SwiftUI where the modifier also matters (I am thinking of the `bold`, `italic`, etc. modifiers on `Text`, which needs to be applied before any `frame` or similar modifiers, that erase the type to `some View`

**Comment by onmyway133** [comment]
Ja I'm fine with Retry being different modifier, or it can just return a Task

File: Documentation/Develop.md [hunk 1]
```diff
@@ -35,73 +35,99 @@ import PumaCore
...
+	Workflow {
+		PrintWorkingDirectory()
+
+		Wait(seconds: 2)
+
+		Retry(times: 2) {
+			PrintWorkingDirectory()
+		}
+
+		RunScript(script: "echo 'Hello Puma'")
```
Comment (onmyway133) [inline Documentation/Develop.md:47]: Can you format these to be 4 spaces (tab with 4 spaces) ?
I think later we need to install SwiftFormat to auto format before commit

Comment (Frederiks96) [reply Documentation/Develop.md:47]: Yes, looks like I missed some files regarding format

**Comment by onmyway133** [comment]
@Frederiks96 Looks good to me now, things are looking up 🚀

**Pull request merged**